lib/src/screens/home/widgets/pending_pointer_listener.dart

```diff
@@ -0,0 +1,43 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+import '../../../models/session.dart';
+import '../../../util/handle_pointer.dart';
+import '../../../widgets/irma_repository_provider.dart';
+
+class PendingPointerListener extends StatefulWidget {
+  final Widget child;
+  const PendingPointerListener({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  @override
+  _PendingPointerListenerState createState() => _PendingPointerListenerState();
+}
+
+class _PendingPointerListenerState extends State<PendingPointerListener> {
+  StreamSubscription<Pointer?>? _pointerSubscription;
+
+  @override
+  void initState() {
+    super.initState();
+
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      final repo = IrmaRepositoryProvider.of(context);
+      _pointerSubscription = repo.getPendingPointer().listen((Pointer? pointer) {
+        if (pointer != null) handlePointer(Navigator.of(context), pointer);
+      });
+    });
```

Comment:
Are we sure this also works when the app becomes active from background? I'd say that `initState` is not called then.

Reply:
Yes this still works. The `PendingPointerListener` is wrapping the inner part of the `HomeScreen` which is always present in the stack (except during enrolment). `initState` only sets up the subscription and provides the callback so after that it just keeps listening to incoming pointers and executing the callback until the lifecycle of the widget ends.